nevergrad/functions/gym/gym_multi.py

```diff
@@ -0,0 +1,777 @@
+        """Applies a neural net parametrized by x to an observation o. Returns an action or logits of actions."""
+        o = o.ravel()
+        if self.control == "linear":
+            # The linear case is simplle.
+            output = np.matmul(o, x[1:, :])
+            output += x[0]
+            return output.reshape(self.output_shape), np.zeros(0)
+        if "structured" not in self.control:
+            # If not structured then we split into two matrices.
+            first_matrix = x[: self.first_size].reshape(self.first_layer_shape) / np.sqrt(len(o))
+            second_matrix = x[self.first_size : (self.first_size + self.second_size)].reshape(
+                self.second_layer_shape
+            ) / np.sqrt(self.num_neurons)
+        else:
+            # In the structured case we should have two entries with the right shapes.
+            assert len(x) == 2
+            first_matrix = np.asarray(x[0][0])
+            second_matrix = np.asarray(x[0][1])
+            assert (
+                first_matrix.shape == self.first_layer_shape
+            ), f"{first_matrix} does not match {self.first_layer_shape}"
+            assert (
+                second_matrix.shape == self.second_layer_shape
+            ), f"{second_matrix} does not match {self.second_layer_shape}"
+        assert len(o) == len(first_matrix[1:]), f"{o.shape} coming in matrix of shape {first_matrix.shape}"
+        output = np.matmul(o, first_matrix[1:])
+        if "deep" in self.control:
+            # The deep case must be split into several layers.
+            current_index = self.first_size + self.second_size
+            internal_layer_size = self.num_neurons ** 2
+            s = (self.num_neurons, self.num_neurons)
+            for _ in range(self.num_internal_layers):
+                output = np.tanh(output)
+                output = np.matmul(
+                    output, x[current_index : current_index + internal_layer_size].reshape(s)
+                ) / np.sqrt(self.num_neurons)
+                current_index += internal_layer_size
+            assert current_index == len(x)
+        output = np.matmul(np.tanh(output + first_matrix[0]), second_matrix)
+        return output[self.memory_len :].reshape(self.output_shape), output[: self.memory_len]
+
+    def gym_multi_function(
+        self, x: np.ndarray, limited_fidelity: bool = False, compiler_gym_pb_index: tp.Optional[int] = None
+    ) -> float:
+        """Do a simulation with parametrization x and return the result.
+
+        Parameters:
+            limited_fidelity: bool
+                whether we use a limited version for the beginning of the training.
+            compiler_gym_pb_index: int or None.
+                index of the compiler_gym pb (<0 for csmith)
+        """
+        # Deterministic conformant: do  the average of 7 simullations always with the same seed.
+        # Otherwise: apply a random seed and do a single simulation.
```

Comment:
I think most people would expect this kind of comments to be related to the code just afterwards. Maybe put it at the end of the docstring instead and say that deterministic conformant corresponds to self.control == "conformant"